sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala

```diff
@@ -1862,7 +1864,24 @@ private[lf] object SBuiltinFun {
 
       val time = getSTimestamp(args, 0)
 
-      machine.needTime(now => Control.Value(SBool(now < time)))
+      machine.needTime(now => {
```

Comment:
`needTime` calls `setDependsOnTime` and so always sets `timeBoundary` to be `Time.Range(now, now)` (where `now` is the current ledger time) prior to `SBULedgerTimeLT` running its time boundary updating code.